authors/apps/authentication/views.py

```diff
@@ -286,3 +288,91 @@ def post(self, request):
         message = {'message': 'New verification token created. Please proceed to your email ' + # noqa
                    user_email + ' to verify your account.'}
         return Response(message, status=status.HTTP_201_CREATED)
+
+
+class PasswordResetView(APIView):
+    """
+    post:
+        Get a user's email where password reset link will be sent.
+    """
+    def post(self, request):
+        data = request.data.get('payload')
+        serializer = PasswordResetSerializer(data=data)
+        serializer.is_valid(raise_exception=True)
+        user_email = data['email']
+        callback_url = data['callback_url']
+        message = "A password reset link has been sent " \
+                  "to your email account. " \
+                  "If you don't receive an email, and " \
+                  "it's not in your spam folder this could mean " \
+                  "you signed up with a different address."
+        try:
+            user = get_object_or_404(User, email=user_email)
+            user_id = user.id
+            payload = {
+
+                "email": user_email,
+                "callback_url": callback_url
+            }
+            token = TokenHandler().create_verification_token(payload)
+            token_data = {
+                "user": user_id,
+                "token": token
+            }
+            serializer = PasswordResetTokenSerializer(data=token_data)
+            serializer.is_valid(raise_exception=True)
+            serializer.save()
+            TokenHandler().send_password_reset_link(user_email,
+                                                    token, callback_url)
+            return Response({"message": message},
+                            status=status.HTTP_200_OK)
+        except Http404:
+            raise APIException({"message": message})
```

Comment:
I don't think it's a good idea to raise an exception in the view. You could return a Response with the proper status code instead.

Reply:
Thank you for capturing this. I have rectified it. Everything should be working alright now.